test/integration/setup.js

```diff
@@ -1,128 +1,153 @@
+  const DisposableNXMaster = await ethers.getContractFactory('DisposableNXMaster');
+  const DisposableMemberRoles = await ethers.getContractFactory('DisposableMemberRoles');
+  const DisposableTokenController = await ethers.getContractFactory('DisposableTokenController');
+  const DisposableProposalCategory = await ethers.getContractFactory('DisposableProposalCategory');
+  const DisposableGovernance = await ethers.getContractFactory('DisposableGovernance');
+  const DisposablePooledStaking = await ethers.getContractFactory('DisposablePooledStaking');
+  const DisposableGateway = await ethers.getContractFactory('DisposableGateway');
+  const DisposableAssessment = await ethers.getContractFactory('DisposableAssessment');
+  const DisposableYieldTokenIncidents = await ethers.getContractFactory('DisposableYieldTokenIncidents');
+  const DisposableIndividualClaims = await ethers.getContractFactory('DisposableIndividualClaims');
 
   // target contracts
-  const NXMaster = artifacts.require('NXMaster');
-  const MemberRoles = artifacts.require('MemberRoles');
-  const TokenController = artifacts.require('TokenController');
-  const ProposalCategory = artifacts.require('ProposalCategory');
-  const Governance = artifacts.require('Governance');
-  const PooledStaking = artifacts.require('LegacyPooledStaking');
-  const Gateway = artifacts.require('LegacyGateway');
-  const YieldTokenIncidents = artifacts.require('YieldTokenIncidents');
-  const IndividualClaims = artifacts.require('IndividualClaims');
-  const Assessment = artifacts.require('Assessment');
+  const NXMaster = await ethers.getContractFactory('NXMaster');
+  const MemberRoles = await ethers.getContractFactory('MemberRoles');
+  const TokenController = await ethers.getContractFactory('TokenController');
+  const ProposalCategory = await ethers.getContractFactory('ProposalCategory');
+  const Governance = await ethers.getContractFactory('Governance');
+  const PooledStaking = await ethers.getContractFactory('LegacyPooledStaking');
+  const Gateway = await ethers.getContractFactory('LegacyGateway');
+  const YieldTokenIncidents = await ethers.getContractFactory('YieldTokenIncidents');
+  const IndividualClaims = await ethers.getContractFactory('IndividualClaims');
+  const Assessment = await ethers.getContractFactory('Assessment');
 
   const signers = await ethers.getSigners();
   const ethersAccounts = getAccounts(signers);
 
   // external
-  const WETH9 = artifacts.require('WETH9');
-  const CSMockSettlement = artifacts.require('CSMockSettlement');
-  const CSMockVaultRelayer = artifacts.require('CSMockVaultRelayer');
+  const WETH9 = await ethers.getContractFactory('WETH9');
+  const CSMockSettlement = await ethers.getContractFactory('CSMockSettlement');
+  const CSMockVaultRelayer = await ethers.getContractFactory('CSMockVaultRelayer');
+
+  const deployImmutable = async (contractFactory, constructorArgs = [], options = {}) => {
+    const { libraries } = options;
```

Comment:
The destructuring is not needed here. Rename `options` to `overrides` and pass it directly without the `if (libraries) ` condition.